pkg/cloudprovider/aws/factory.go

```diff
@@ -86,9 +87,24 @@ func (f *Factory) CapacityFor(provisioner *v1alpha1.Provisioner) cloudprovider.C
 	}
 }
 
+// Checks for env vars in the defined order and if none of those are set falls back to using EC2
+// IMDS for region lookup if neither AWS_DEFAULT_REGION or AWS_REGION is defined.
 func withRegion(sess *session.Session) *session.Session {
+	if region, envAwsDefaultRegion := os.LookupEnv("AWS_DEFAULT_REGION"); envAwsDefaultRegion {
+		log.PrettyInfof("Found AWS_DEFAULT_REGION env var set, using: %v", region)
+		sess.Config.Region = aws.String(region)
+		return sess
+	}
+
+	if region, envAwsRegion := os.LookupEnv("AWS_REGION"); envAwsRegion {
+		log.PrettyInfof("Found AWS_REGION env var set, using: %v", region)
+		sess.Config.Region = aws.String(region)
+		return sess
+	}
+
 	region, err := ec2metadata.New(sess).Region()
 	log.PanicIfError(err, "failed to call the metadata server's region API")
+	log.PrettyInfof("Retrieved region %v from EC2 Instance Metadata", region)
```

Comment:
I'd love if we had parity for these log lines. We also tend to do debug statements for the cloud provider to avoid polluting the logs.
```
zap.S().Debug("Discovered region %s from env['AWS_DEFAULT_REGION']")
zap.S().Debug("Discovered region %s from IMDS")
```